k8s/upgrades/0.8.2-0.9.0/cstor/cstor-pool-update-082-090.yaml

```diff
@@ -0,0 +1,664 @@
+    disable: false
+  post: |
+    {{- jsonpath .JsonResult "{.metadata.labels.openebs\\.io/version}" | trim | saveAs "StoragePool.labels" .TaskResult | noop -}}
+
+    {{- $message := printf "Successfully got details of StoragePool {%s}." .UpgradeItem.name -}}
+    {{- $status :="" }}
+    {{- if eq .TaskResult.StoragePool.labels "0.8.2" }}
+    {{- $status =.Config.successStatus.value }}
+    {{- else }}
+    {{- $status =.Config.failStatus.value }}
+    {{- end }}
+
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-cstor-pool-082-090-get-storagepool" -}}
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace "default" -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+    {{- if eq $status .Config.failStatus.value }}
+    {{- verifyErr "StoragePool version is not 0.8.2" true | saveAs "StoragePool.verifyErr" .TaskResult | noop -}} 
+    {{- end }}
+---
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-patch-upgrade-result
+  namespace: default
+spec:
+  meta: |
+    id: patchResult
+    apiVersion: openebs.io/v1alpha1
+    kind: UpgradeResult
+    action: patch
+    objectName: {{ .UpgradeItem.upgradeResultName }}
+    runNamespace: {{ .UpgradeItem.upgradeResultNamespace }}
+  task: |-
+    type: merge
+    pspec: |-
+      status:
+        resource:
+          name: {{ .UpgradeItem.name }}
+          namespace: {{ .UpgradeItem.namespace }}
+          kind: {{ .UpgradeItem.kind }}
+---
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-cstor-pool-082-090-patch-deployment-image
+  namespace: default
+spec:
+  meta: |
+    {{ $isOldCStorPool := contains .Config.baseversion.value .TaskResult.poolDeployment.cstorpoolimage }}
+    {{ $isOldCStorPoolMGMT := contains .Config.baseversion.value .TaskResult.poolDeployment.cstorpoolmgmtimage  }}
+    {{ $shouldpatch := or $isOldCStorPool $isOldCStorPoolMGMT | toString }}
+    id: patchDeploymentImage
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    action: patch
+    objectName: {{ .UpgradeItem.name }}
+    runNamespace: {{ .UpgradeItem.namespace }}
+    disable: {{ eq $shouldpatch "false" }}
+  task: |-
+    type: strategic
+    pspec: |-
+      spec:
+        template:
+          metadata:
+            labels:
+              openebs.io/version: {{ .Config.targetversion.value }}
+              openebs.io/cstor-pool: {{ .UpgradeItem.name }} 
+          spec:
+            containers:
+            - name: cstor-pool
+              image: quay.io/openebs/cstor-pool:v0.9.x-ci
```

Comment:
can we remove this hardcoding & instead make use of cas config or runtime config